src/configuration/yaml.md

```diff
@@ -0,0 +1,173 @@
+
+[YAML](https://en.wikipedia.org/wiki/YAML) ("YAML Ain't Markup Language") is a human-readable data file format, well suited to human-edited configuration files. Nearly all aspects of your project's build and deploy pipeline are controlled via YAML files.
+
+YAML is a whitespace-sensitive format that is especially good at key/value type configuration, such as that used by Platform.sh. There are many good YAML tutorials online and the format is reasonably self-documenting.  We especially recommend:
+
+* [GravCMS's YAML tutorial](https://learn.getgrav.org/advanced/yaml)
+* [Learn YAML in Y Minutes](https://learnxinyminutes.com/docs/yaml/)
+
+The following is only a cursory look at YAML itself.  The tutorials above will provide a more in-depth introduction.
+
+## Basic YAML
+
+A YAML file is a text file that ends in `.yaml`.  (Some systems use an alternative `.yml` extension, but Platform.sh uses the four-letter extension.)  It consists primarily of key value pairs, and supports nesting.  For example:
+
+```yaml
+name: 'app'
+type: 'php:7.1'
+build:
+    flavor: 'composer'
+
+disk: 1024
+```
+
+This example defines a key `name` with value `app`, a key `type` with value `php:7.1`, a key `disk` with a value 1024, and a key `build` that is itself a nested set of key/value pairs, of which there is only one: `flavor`, whose value is `composer`.  Nested values are often referenced using a dotted syntax, such as `build.flavor`.
+
+Keys are always strings, and may be quoted or not.  Values may be strings, numbers, or booleans.  Alphanumeric strings may be quoted or not.  More complex strings (with punctuation, etc.) must be quoted.  Numbers should not be quoted.  The boolean values `true` and `false` should never be quoted.
+
+In general the order of keys in a YAML file does not matter.  Neither do blank lines.  Indentation may be with any number of spaces, as long as it is consistent throughout the file.  Platform.sh examples by convention use four-space indentation.
+
+## Multi-line strings
+
+In case of long, multi-line strings, the `|` character tells the YAML parser that the following, indented lines are all part of the same string.  That is, this:
+
+```yaml
+hooks:
+    build: |
+        set -e
+        cp a.txt b.txt
+```
+
+creates a nested property `hooks.build`, which has the value `set -e\ncp a.txt b.txt`.  (That is, a string with a line break in it.)  That is useful primarily for hooks, which allow the user to enter small shell scripts within the YAML file.
+
+## Local tags
+
+YAML allows for special "tags" on values that change their meaning.  These tags may be customized for individual applications so may vary from one system to another.  For Platform.sh there are two that are useful to know about.
+
+### !archive
+
+The `!archive` tag specifies that the value it tags is a reference to a directory on disk, relative to the location of the YAML file.  Essentially it defines the value of key as "this entire directory".  For example:
```

Comment:
Let's only document `!include`?

Reply:
Er.  Why?  Anyone using Solr with Psh right now is using !archive.  Not mentioning it seems very obtuse.

Reply:
Or at least mention that it's deprecated? Dunno, if we're superseding `!archive`, not giving it a big place in the doc makes sense to me.